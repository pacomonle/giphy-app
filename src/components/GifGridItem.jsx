import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({title, url, id}) => {
   // console.log(title, url, id)
    return (
        <div className='card animate__animated animate__fadeInBottomRight'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  id: PropTypes.string
  
}

export default GifGridItem
