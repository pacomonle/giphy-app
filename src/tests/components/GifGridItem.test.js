import { shallow } from 'enzyme';
import React from 'react';
import GifGridItem from '../../components/GifGridItem';




describe('Pruebas GifGrid.test.js', () => {
    const title = 'un titulo'
    const url = 'http://localhost:8080'
    const wraper = shallow(<GifGridItem title={title} url={url}></GifGridItem>)

    test('should mostrar valores de <GifGridItem></GifGridItem> ', () => {
       // const title = 'un titulo'
       // const url = 'http://localhost:8080'
       // const wraper = shallow(<GifGridItem title={title} url={url}></GifGridItem>)
        expect(wraper).toMatchSnapshot()
    })

    test('should el title debe de esta en un <p></p>', () => {
        // 1. seleccionamos el parrafo - p
        const parraf = wraper.find('p')
        // 2. comprobar la iguladad
        expect(parraf.text().trim()).toBe(title)
    })

    test('should a la etiqueta <img> comprobar src y alt', () => {
         // 1. seleccionamos la imagen - img
        const image = wraper.find('img')
        // comprobar propiedades
        console.log(image.html())
        console.log(image.props(), image.prop('src'), image.props().alt)
        expect(image.prop('alt')).toBe(title)
        expect(image.props().src).toBe(url)

    })
    
    test('should class animate__fadeInBottomRight  de animate en el div ', () => {
     // 1. seleccionamos el div - div
        const animate = 'animate__fadeInBottomRight'
        const div = wraper.find('div')
        // console.log(props())
        console.log(div.prop('className'))
        const className = div.prop('className')
     // 2. comprobar con un true   
         // para comprobar la negacion .not.toBe
         expect(className.includes(animate)).toBe(true)

        
    })
    
    
})
