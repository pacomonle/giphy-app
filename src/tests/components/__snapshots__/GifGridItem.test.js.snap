// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGrid.test.js should mostrar valores de <GifGridItem></GifGridItem>  1`] = `
<div
  className="card animate__animated animate__fadeInBottomRight"
>
  <img
    alt="un titulo"
    src="http://localhost:8080"
  />
  <p>
    un titulo
  </p>
</div>
`;
